openssl/src/x509/mod.rs

```diff
@@ -433,6 +433,28 @@ impl<'a> X509Ref<'a> {
         }
     }
 
+    /// Returns certificate Not Before validity period.
+    /// Requires the `x509_expiry` feature.
+    #[cfg(feature = "x509_expiry")]
+    pub fn not_after(&self) -> Asn1TimeRef {
```

Comment:
The lifetime is elided - what's actually happening is identical to that patch. I like the explicitness though - can you add the `'b` lifetime in as well? The constructor change isn't necessary.